### ISSUE TITLE
Exception when migrating from Craft 2: Calling unknown method: craft\console\Request::getBodyParam()

### ISSUE THREAD

**Issue by kgi-cwells** [barrelstrength/craft-sprout-forms#321]
### Description
When migrating from Craft 2 to 3.2.5.1, migration fails with the following:
```
2019-07-22 09:34:04 [-][-][-][error][craft\errors\MigrateException] yii\base\UnknownMethodException: Calling unknown method: craft\console\Request::getBodyParam() in /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Component.php:300
Stack trace:
#0 /shared/httpd/klipsch3/vendor/barrelstrength/sprout-base-email/src/elements/db/NotificationEmailQuery.php(47): yii\base\Component->__call('getBodyParam', Array)
#1 /shared/httpd/klipsch3/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1285): barrelstrength\sproutbaseemail\elements\db\NotificationEmailQuery->beforePrepare()
#2 /shared/httpd/klipsch3/vendor/yiisoft/yii2/db/QueryBuilder.php(227): craft\elements\db\ElementQuery->prepare(Object(craft\db\mysql\QueryBuilder))
#3 /shared/httpd/klipsch3/vendor/yiisoft/yii2/db/Query.php(146): yii\db\QueryBuilder->build(Object(barrelstrength\sproutbaseemail\elements\db\NotificationEmailQuery))
#4 /shared/httpd/klipsch3/vendor/yiisoft/yii2/db/Query.php(421): yii\db\Query->createCommand(Object(craft\db\Connection))
#5 /shared/httpd/klipsch3/vendor/craftcms/cms/src/db/Query.php(224): yii\db\Query->exists(NULL)
#6 /shared/httpd/klipsch3/vendor/craftcms/cms/src/elements/db/ElementQuery.php(1469): craft\db\Query->exists(NULL)
#7 /shared/httpd/klipsch3/vendor/craftcms/cms/src/base/Element.php(1639): craft\elements\db\ElementQuery->exists()
#8 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Component.php(139): craft\base\Element->getHasDescendants()
#9 /shared/httpd/klipsch3/vendor/craftcms/cms/src/base/Element.php(879): yii\base\Component->__get('hasDescendants')
#10 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/ArrayableTrait.php(126): craft\base\Element->__get('hasDescendants')
#11 /shared/httpd/klipsch3/vendor/craftcms/cms/src/web/View.php(515): yii\base\Model->toArray(Array, Array, false)
#12 /shared/httpd/klipsch3/vendor/craftcms/cms/src/helpers/ElementHelper.php(184): craft\web\View->renderObjectTemplate('sprout-forms/{{...', Object(barrelstrength\sproutbaseemail\elements\NotificationEmail), Array)
#13 /shared/httpd/klipsch3/vendor/craftcms/cms/src/helpers/ElementHelper.php(102): craft\helpers\ElementHelper::_renderUriFormat('sprout-forms/{s...', Object(barrelstrength\sproutbaseemail\elements\NotificationEmail))
#14 /shared/httpd/klipsch3/vendor/craftcms/cms/src/validators/ElementUriValidator.php(54): craft\helpers\ElementHelper::setUniqueUri(Object(barrelstrength\sproutbaseemail\elements\NotificationEmail))
#15 /shared/httpd/klipsch3/vendor/yiisoft/yii2/validators/Validator.php(257): craft\validators\ElementUriValidator->validateAttribute(Object(barrelstrength\sproutbaseemail\elements\NotificationEmail), 'uri')
#16 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Model.php(367): yii\validators\Validator->validateAttributes(Object(barrelstrength\sproutbaseemail\elements\NotificationEmail), Array)
#17 /shared/httpd/klipsch3/vendor/barrelstrength/sprout-base-email/src/services/NotificationEmails.php(44): yii\base\Model->validate(Array, false)
#18 /shared/httpd/klipsch3/vendor/barrelstrength/sprout-forms/src/migrations/m180314_161540_craft2_to_craft3.php(162): barrelstrength\sproutbaseemail\services\NotificationEmails->saveNotification(Object(barrelstrength\sproutbaseemail\elements\NotificationEmail))
#19 /shared/httpd/klipsch3/vendor/craftcms/cms/src/db/Migration.php(56): barrelstrength\sproutforms\migrations\m180314_161540_craft2_to_craft3->safeUp()
#20 /shared/httpd/klipsch3/vendor/craftcms/cms/src/db/MigrationManager.php(243): craft\db\Migration->up(true)
#21 /shared/httpd/klipsch3/vendor/craftcms/cms/src/db/MigrationManager.php(163): craft\db\MigrationManager->migrateUp(Object(barrelstrength\sproutforms\migrations\m180314_161540_craft2_to_craft3))
#22 /shared/httpd/klipsch3/vendor/craftcms/cms/src/services/Updates.php(225): craft\db\MigrationManager->up()
#23 /shared/httpd/klipsch3/vendor/craftcms/cms/src/console/controllers/MigrateController.php(259): craft\services\Updates->runMigrations(Array)
#24 [internal function]: craft\console\controllers\MigrateController->actionAll()
#25 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#26 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#27 /shared/httpd/klipsch3/vendor/yiisoft/yii2/console/Controller.php(148): yii\base\Controller->runAction('all', Array)
#28 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction('all', Array)
#29 /shared/httpd/klipsch3/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#30 /shared/httpd/klipsch3/vendor/craftcms/cms/src/console/Application.php(93): yii\console\Application->runAction('migrate/all', Array)
#31 /shared/httpd/klipsch3/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#32 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#33 /shared/httpd/klipsch3/craft(22): yii\base\Application->run()
#34 {main}

Next craft\errors\MigrationException: An error occurred while executing the "barrelstrength\sproutforms\migrations\m180314_161540_craft2_to_craft3 migration: Calling unknown method: craft\console\Request::getBodyParam() in /shared/httpd/klipsch3/vendor/craftcms/cms/src/db/MigrationManager.php:260
Stack trace:
#0 /shared/httpd/klipsch3/vendor/craftcms/cms/src/db/MigrationManager.php(163): craft\db\MigrationManager->migrateUp(Object(barrelstrength\sproutforms\migrations\m180314_161540_craft2_to_craft3))
#1 /shared/httpd/klipsch3/vendor/craftcms/cms/src/services/Updates.php(225): craft\db\MigrationManager->up()
#2 /shared/httpd/klipsch3/vendor/craftcms/cms/src/console/controllers/MigrateController.php(259): craft\services\Updates->runMigrations(Array)
#3 [internal function]: craft\console\controllers\MigrateController->actionAll()
#4 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#5 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#6 /shared/httpd/klipsch3/vendor/yiisoft/yii2/console/Controller.php(148): yii\base\Controller->runAction('all', Array)
#7 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction('all', Array)
#8 /shared/httpd/klipsch3/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#9 /shared/httpd/klipsch3/vendor/craftcms/cms/src/console/Application.php(93): yii\console\Application->runAction('migrate/all', Array)
#10 /shared/httpd/klipsch3/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#11 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#12 /shared/httpd/klipsch3/craft(22): yii\base\Application->run()
#13 {main}

Next craft\errors\MigrateException: An error occurred while migrating Sprout Forms. in /shared/httpd/klipsch3/vendor/craftcms/cms/src/services/Updates.php:233
Stack trace:
#0 /shared/httpd/klipsch3/vendor/craftcms/cms/src/console/controllers/MigrateController.php(259): craft\services\Updates->runMigrations(Array)
#1 [internal function]: craft\console\controllers\MigrateController->actionAll()
#2 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array(Array, Array)
#3 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Controller.php(157): yii\base\InlineAction->runWithParams(Array)
#4 /shared/httpd/klipsch3/vendor/yiisoft/yii2/console/Controller.php(148): yii\base\Controller->runAction('all', Array)
#5 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Module.php(528): yii\console\Controller->runAction('all', Array)
#6 /shared/httpd/klipsch3/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction('migrate/all', Array)
#7 /shared/httpd/klipsch3/vendor/craftcms/cms/src/console/Application.php(93): yii\console\Application->runAction('migrate/all', Array)
#8 /shared/httpd/klipsch3/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction('migrate/all', Array)
#9 /shared/httpd/klipsch3/vendor/yiisoft/yii2/base/Application.php(386): yii\console\Application->handleRequest(Object(craft\console\Request))
#10 /shared/httpd/klipsch3/craft(22): yii\base\Application->run()
#11 {main}
```

### Steps to reproduce

1. Migrate from Craft 2.7.9 / Sprout Forms 2.6.4 to Craft 3.2.5.1 / Sprout Forms 3.3.7

### Additional info

- Craft version: 3.2.5.1
- PHP version: 7.1.29
- Database driver & version:  MySQL 5.0.12
- Plugins & versions: Sprout Forms 3.3.7

**Comment by kgi-cwells** [comment]
Update on this. This seems to happen only when attempting to do the update via the command line, specifically when doing a `./craft migrate/all`. When visiting the CP and having it run the update from there, it seems to work fine (assuming because at that point it's not using the console API, which is why that error above happens)

Hope this helps you solve the problem.

**Comment by BenParizek** [comment]
@kgi-cwells This should be resolved in the latest release of Sprout Forms: v3.4.0
